fuse/fuse_test.go

```diff
@@ -17,104 +19,290 @@ import (
+		return &nothing{}, err
 	}
+	bfInfo.fuseMount = fuseMount
+	return &nothing{}, err
 }
 
-func withMountFromFs(t *testing.T, opts MountOptions, fs *catfs.FS, f func(mount *Mount)) {
-	mntPath := filepath.Join(os.TempDir(), "brig-fuse-mountdir")
+func (bmh *brigHelp) unmountFuseAndcloseDummyBrigFS(ctx context.Context, req nothing) (*nothing, error) {
+	defer os.RemoveAll(bfInfo.fuseMount.Dir)
+	defer os.RemoveAll(bfInfo.dbPath)
+	// first unmount fuse directory
+	if err := lazyUnmount(bfInfo.fuseMount.Dir); err != nil {
+		skipableErr := "exit status 1: fusermount: entry for " + bfInfo.fuseMount.Dir + " not found in /etc/mtab"
+		log.Debug(skipableErr)
+		if err.Error() != skipableErr {
+			return &nothing{}, err
+		}
+	}
 
+	// now close brig FS
+	err := bfInfo.bfs.Close()
+	if err != nil {
+		log.Fatalf("Could not close brig filesystem: %v", err)
+	}
+	return &nothing{}, err
+}
+
+func makeFuseMount(bfs *catfs.FS, mntPath string, opts MountOptions) (*Mount, error) {
 	// Make sure to unmount any mounts that are there.
 	// Possibly there are some leftovers from previous failed runs.
-	lazyUnmount(mntPath)
+	if err := lazyUnmount(mntPath); err != nil {
+		skipableErr := "exit status 1: fusermount: entry for " + mntPath + " not found in /etc/mtab"
+		log.Debug(skipableErr)
+		if err.Error() != skipableErr {
+			return nil, err
+		}
+	}
 
 	if err := os.MkdirAll(mntPath, 0777); err != nil {
-		t.Fatalf("Unable to create empty mount dir: %v", err)
+		log.Fatalf("Unable to create empty mount dir: %v", err)
+		return nil, err
 	}
 
-	defer testutil.Remover(t, mntPath)
-
-	mount, err := NewMount(fs, mntPath, nil, opts)
+	mount, err := NewMount(bfs, mntPath, nil, opts)
 	if err != nil {
-		t.Fatalf("Cannot create mount: %v", err)
+		log.Fatalf("Cannot create mount: %v", err)
+		return nil, err
 	}
+	return mount, err
+}
+
+type brigPayload struct {
+	Path string
+	Data []byte
+}
+
+func (bmh *brigHelp) brigStage(ctx context.Context, req brigPayload) (*nothing, error) {
+	err := bfInfo.bfs.Stage(req.Path, bytes.NewReader(req.Data))
+	return &nothing{}, err
+}
 
-	f(mount)
+// Get data from a file stored by brig fs
+func (bmh *brigHelp) brigGetData(ctx context.Context, req brigPayload) (*brigPayload, error) {
+	out := brigPayload{}
+	out.Path = req.Path
 
-	if err := mount.Close(); err != nil {
-		t.Fatalf("Closing mount failed: %v", err)
+	stream, err := bfInfo.bfs.Cat(req.Path)
+	if err != nil {
+		log.Fatalf("Could not get stream for a brig file: %v", err)
+		return &out, err
 	}
+	result := bytes.NewBuffer(nil)
+	_, err = stream.WriteTo(result)
+	if err != nil {
+		log.Fatalf("Streaming to a buffer failed: %v", err)
+		return &out, err
+	}
+	out.Data = result.Bytes()
+
+	return &out, err
 }
 
-func withMount(t *testing.T, opts MountOptions, f func(mount *Mount)) {
-	withDummyFS(t, func(fs *catfs.FS) {
-		withMountFromFs(t, opts, fs, f)
-	})
+var helpers spawntest.Registry
+var brigHelper = helpers.Register("brigHelp", &brigHelp{})
```

Comment:
Prefixing names with the program name is rather unusual in Go, I suggest to use more specific names like ``fuseHelper`` or something.

Reply:
My bad: `brigFS` is officially `catfs`. But I did this mental substition through the `fuse-test` code.  Why do you call it `cat...`, is it short for catalog?

But I see why it would be confusing, for those who read the code. I will relabel the problem spots.

Reply:
It stands for **c**ontent **a**ddressed **t**ransactional **f**ile**s**ystem. But that's just alibi, I just like cats.

Reply:
addressed with massive renaming in f33b437